File: R/8kyu/8kyu-short-long-short.r
# Short Long Short
# Language: R
# Version 3.4.1
# Difficulty: 8 kyu

solution <- function(a, b) {
  ifelse(nchar(a) < nchar(b), paste0(a, b, a), paste0(b, a, b))
}


# Tests
library(stringr)

test_that("example test cases", {
   expect_equal(solution('45', '1'), '1451')
   expect_equal(solution('13', '200'), '1320013')
   expect_equal(solution('Soon', 'Me'), 'MeSoonMe')
   expect_equal(solution('U', 'False'), 'UFalseU')
})

File: Python/8kyu/8kyu-short-long-short.py
# Short Long Shor
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

def solution(a, b):
    return a + b + a if len(a) < len(b) else b + a + b


# Tests
import codewars_test as test
from solution import solution

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(solution('45', '1'), '1451')
        test.assert_equals(solution('13', '200'), '1320013')
        test.assert_equals(solution('Soon', 'Me'), 'MeSoonMe')
        test.assert_equals(solution('U', 'False'), 'UFalseU')